### ISSUE TITLE
Enumerations

### ISSUE THREAD

**Issue by eed3si9n** [tree-sitter/tree-sitter-scala#72]
https://docs.scala-lang.org/scala3/reference/enums/enums.html

```scala
enum Option[+T]:
  case Some(x: T) extends Option[T]
  case None       extends Option[Nothing]

enum Color(val rgb: Int):
  case Red   extends Color(0xFF0000)
  case Green extends Color(0x00FF00)
  case Blue  extends Color(0x0000FF)
  case Mix(mix: Int) extends Color(mix)
```

https://docs.scala-lang.org/scala3/reference/contextual/derivation.html

```scala
enum Tree[T] derives Eq, Ordering, Show:
  case Branch(left: Tree[T], right: Tree[T])
  case Leaf(elem: T)
```

**Comment by keynmol** [comment]
Starting point can be copied from https://github.com/keynmol/tree-sitter-scala/pull/1/commits/df196cbec53717f7c1146156c63d90966e68f58e

**Comment by keynmol** [comment]
I think with fixes in #117 this can be closed as done
